src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -495,6 +495,13 @@ private function getShippingDisplayValue($cart, $shippingCost)
     {
         $shippingDisplayValue = '';
 
+        // if one of the applied cart rules have free shipping, then the shipping display value is 'Free'
+        foreach ($cart->getCartRules() as $rule) {
```

Comment:
Thank you for addressing this issue. I am just a bit confused in front of the the `else` statement in line 507. Doesn't it contain part/or whole of the logic you added? Again thanks for your time

Reply:
Ping @matthieu-rolland